src/cmd/ksh93/tests/builtins.sh

```diff
@@ -986,7 +986,10 @@ EOF
 
 # ======
 # Builtins should handle unrecognized options correctly
-while IFS= read -r bltin <&3
+# Note: To workaround https://github.com/ksh93/ksh/issues/165, put the list
+# of builtins in a file, then read from that.
+builtin > "$tmp/builtin-list"
+while IFS= read -r bltin
```

Comment:
Some built-ins may read from standard input, so this test should continue to use file descriptor 3 (for the `read` and the input redirection following `done`).